src/main/java/aRC/Parser.java

```diff
@@ -0,0 +1,243 @@
+                    Arrays.copyOfRange(commandArgs, indexOfBy + 1, commandArgs.length));
+
+            if (title == "") {
+                throw new EmptyTitleException();
+            } else if (deadline == "") {
+                throw new InvalidArgumentException();
+            } else {
+                LocalDate ld = this.validateDateTime(deadline);
+                this.taskList.addTask(new Deadline(title, false, ld));
+                this.storage.save(this.taskList);
+            }
+        }
+    }
+
+    /**
+     * Parses the event command
+     * @param commandArgs Array of Strings representing command arguments
+     * @throws DukeException Throws a aRC.DukeException specific to this program
+     */
+    public void parseEvent(String[] commandArgs) throws DukeException {
+        if (!Arrays.asList(commandArgs).contains("/at")) {
+            throw new InvalidArgumentException();
+        } else {
+            int indexOfBy = Arrays.asList(commandArgs).indexOf("/at");
+            String title = String.join(" ", Arrays.copyOfRange(commandArgs, 0, indexOfBy));
+            String time = String.join(" ",
+                    Arrays.copyOfRange(commandArgs, indexOfBy + 1, commandArgs.length));
+
+            if (title == "") {
+                throw new EmptyTitleException();
+            } else if (time == "") {
+                throw new InvalidArgumentException();
+            } else {
+                this.taskList.addTask(new Event(title, false, time));
+                this.storage.save(this.taskList);
+            }
+        }
+    }
+
+    /**
+     * Parses the delete command
+     * @param commandArgs Array of Strings representing command arguments
+     * @throws DukeException Throws a aRC.DukeException specific to this program
+     */
+    public void parseDelete(String[] commandArgs) throws DukeException {
+        if (commandArgs.length != 1) {
+            throw new InvalidArgumentException();
+        } else {
+            int index = this.validateIndex(commandArgs[0]) - 1;
+            this.taskList.deleteTask(index);
+            this.storage.save(this.taskList);
+        }
+    }
+
+    /**
+     * Parses the find command
+     * @param commandArgs Array of Strings representing command arguments
+     * @throws InvalidArgumentException If no arguments are entered
+     */
+    public void parseFind(String[] commandArgs) throws InvalidArgumentException {
+        if (commandArgs.length == 0) {
+            throw new InvalidArgumentException();
+        } else {
+            String keyword = String.join(" ", commandArgs);
+            this.taskList.listTasks(keyword);
+        }
+    }
+
+    /**
+     * Validates a given index String
+     * @param index The index String
+     * @return The integer index if it is valid
+     * @throws InvalidArgumentException If the index String is invalid
+     */
+    public int validateIndex(String index) throws InvalidArgumentException {
```

Comment:
I really liked how you separated the checks into a separate. Makes the entire implementation neater and more readable 👍